mac

```diff
@@ -166,29 +166,62 @@ brew_install_or_upgrade 'hub'
 # shellcheck disable=SC2016
 append_to_file "$HOME/.zshrc" 'eval "$(hub alias -s)"'
 
+
+fancy_echo 'Checking on Node.js installation...'
+
 if ! brew_is_installed "node"; then
-  if ! command -v n > /dev/null; then
-    fancy_echo 'Installing n and latest Node.js and NPM...'
-    curl -L http://git.io/n-install | bash -s -- -y latest
+  if command -v nvm > /dev/null; then
+    if ! command -v n > /dev/null; then
+      fancy_echo 'Installing n and latest Node.js and NPM...'
+      curl -L http://git.io/n-install | bash -s -- -y latest
+    else
+      fancy_echo 'Updating n...'
+      n-update -y
+    fi
   else
-    fancy_echo 'Updating n...'
-    n-update -y
+    fancy_echo 'nvm detected.  Skipping...'
   fi
+else
+  brew_install_or_upgrade 'node'
 fi
 
-brew_install_or_upgrade 'python3'
+fancy_echo '...Finished Node.js installation checks.'
+
+
+fancy_echo 'Checking on Python installation...'
+
+if ! brew_is_installed "python3"; then
+  brew_install_or_upgrade 'pyenv'
+  brew_install_or_upgrade 'pyenv-virtualenv'
+  brew_install_or_upgrade 'pyenv-virtualenvwrapper'
+
+  # pyenv currently doesn't have a convenience version to use, e.g., "latest"
+  if ! pyenv versions | ag "3.5.1" > /dev/null; then
```

Comment:
Yeah I'm not a fan of having to maintain this either, which is difficult given that `pyenv` doesn't support any convenience alias like that (at least not yet).  I've noticed that the latest versions do make it in right away, but you need to know which specific version.